app/controllers/protocols_controller.rb

```diff
@@ -658,194 +822,286 @@ def export
+      end
     end
   end
 
   def create_new_modal
     @new_protocol = Protocol.new
     respond_to do |format|
-      format.json {
+      format.json do
         render json: {
-          html: render_to_string({
-            partial: "protocols/index/create_new_modal_body.html.erb"
-          })
+          html: render_to_string(partial: 'protocols/index/create_new_modal_body.html.erb')
         }
-      }
+      end
     end
   end
 
   def edit_name_modal
     respond_to do |format|
-      format.json {
+      format.json do
         render json: {
           title: I18n.t('protocols.header.edit_name_modal.title',
                         protocol: escape_input(@protocol.name)),
-          html: render_to_string({
-            partial: "protocols/header/edit_name_modal_body.html.erb"
-          })
+          html: render_to_string(partial: 'protocols/header/edit_name_modal_body.html.erb')
         }
-      }
+      end
     end
   end
 
   def edit_keywords_modal
     respond_to do |format|
-      format.json {
+      format.json do
         render json: {
           title: I18n.t('protocols.header.edit_keywords_modal.title',
                         protocol: escape_input(@protocol.name)),
-          html: render_to_string({
-            partial: "protocols/header/edit_keywords_modal_body.html.erb"
-          }),
+          html: render_to_string(partial: 'protocols/header/edit_keywords_modal_body.html.erb'),
           keywords: @protocol.team.protocol_keywords_list
         }
-      }
+      end
     end
   end
 
   def edit_authors_modal
     respond_to do |format|
-      format.json {
+      format.json do
         render json: {
           title: I18n.t('protocols.header.edit_authors_modal.title',
                         protocol: escape_input(@protocol.name)),
-          html: render_to_string({
-            partial: "protocols/header/edit_authors_modal_body.html.erb"
-          })
+          html: render_to_string(partial: 'protocols/header/edit_authors_modal_body.html.erb')
         }
-      }
+      end
     end
   end
 
   def edit_description_modal
     respond_to do |format|
-      format.json {
+      format.json do
         render json: {
           title: I18n.t('protocols.header.edit_description_modal.title',
                         protocol: escape_input(@protocol.name)),
-          html: render_to_string({
-            partial: "protocols/header/edit_description_modal_body.html.erb"
-          })
+          html: render_to_string(partial: 'protocols/header/edit_description_modal_body.html.erb')
         }
-      }
+      end
     end
   end
 
   private
 
+  def protocolsio_step_description_populate(result, iterating_key, pos2)
+    if !iterating_key['data'].nil? && iterating_key['data'] != '' &&
+       result['steps'][pos2.to_s]['description']
+      append = '<br>' + iterating_key['data'] + '<br>'
+      result['steps'][pos2.to_s]['description'] << append
+    elsif !result['steps'][pos2.to_s]['description']
```

Comment:
What is the difference between first and second if case?

Reply:
in the first one i use the shovel operator to concatenate the append variable to the result description hash, in the second one, i just asign it the data if the result description hash is null, since then i cant concatenate.